### PR TITLE
deployments: deploy on Goerli testnet

### LINKED ISSUE #1257: Deploy contracts on goerli
# Rationale
Since many of the rollups stopped maintaining their rinkeby deployments, we ended up having a broken testnet environment at the moment. This is only one reason we need to start moving to Goerli.

Another reason for this move is that we will be able to test our upcoming contract v1 implementation as well as all the other components (**relayer**, subgraph, frontend, etc.) before going live on mainnet with the new contract version & arbitrum support. 

On Goerli, we have [Optimism](https://dev.optimism.io/kovan-to-goerli/), [Arbitrum](https://medium.com/offchainlabs/new-g%C3%B6rli-testnet-and-getting-rinkeby-ready-for-nitro-3ff590448053), [zkSync](https://medium.com/@CalendarDefi/testnet-guide-zksync-2-0-6fabedf44f21) and probably more to follow.

This can be considered a part of https://github.com/beamer-bridge/beamer/issues/953 as a minimal version of it.

# Acceptance criteria

- [x] Create Goerli deployment config
- [x] Deploy on Goerli
- [x] Deploy on Goerli Arbitrum
- [x] Deploy on OP Arbitrum?(not sure if needed)

### PR DESCRIPTION
**Pull request by GabrielBuragev**
closes #1257

### CODE REVIEW

File: scripts/deployment/goerli-testnets.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "L1": {
+    "name": "Goerli",
+    "rpc": "https://eth-goerli.g.alchemy.com/v2/lk5Vo1HAubyYMairTVMtKHj0ynpulnBb",
```
Comment (istankovic) [inline scripts/deployment/goerli-testnets.json:4]: Are we ok with storing our Alchemy endpoints in public like this? @fredo ?

Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json:4]: Sorry, forgot my alchemy endpoints here. Will swap them with the public rpc endpoints.

Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json]: ✅

File: scripts/deployment/goerli-testnets.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "name": "Arbitrum Goerli",
+      "rpc": "https://goerli-rollup.arbitrum.io/rpc",
+      "chain_id": 421613,
+      "l1_messenger": [
+        "ArbitrumL1Messenger",
+        "0xaf4159A80B6Cc41ED517DB1c453d1Ef5C2e4dB72",
+        "0x6BEbC4925716945D46F0Ec336D5C2564F419682C"
+      ],
+      "l2_messenger": "ArbitrumL2Messenger",
+      "finality_period": 604800,
```
Comment (istankovic) [inline scripts/deployment/goerli-testnets.json:18]: Just to make sure: is this finality period correct?

File: scripts/deployment/goerli-testnets.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    {
+      "name": "Optimism Goerli",
+      "rpc": "https://goerli.optimism.io",
+      "chain_id": 420,
+      "l1_messenger": [
+        "OptimismL1Messenger",
+        "0x5086d1eEF304eb5284A0f6720f79403b4e9bE294"
+      ],
+      "l2_messenger": "OptimismL2Messenger",
+      "finality_period": 604800,
```
Comment (istankovic) [inline scripts/deployment/goerli-testnets.json:35]: Just to make sure: is this finality period correct?

File: scripts/deployment/goerli-testnets.json (hunk 2 above)
Comment (fredo) [reply scripts/deployment/goerli-testnets.json:18]: However, we need to take care that we don't get below the actual finality period

File: scripts/deployment/goerli-testnets.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        "0xaf4159A80B6Cc41ED517DB1c453d1Ef5C2e4dB72",
+        "0x6BEbC4925716945D46F0Ec336D5C2564F419682C"
+      ],
+      "l2_messenger": "ArbitrumL2Messenger",
+      "finality_period": 604800,
+      "request_manager_arguments": {
+        "claim_stake": 0.00047,
+        "claim_request_extension": 3600,
+        "claim_period": 3600,
+        "challenge_period_extension": 3600
```
Comment (fredo) [inline scripts/deployment/goerli-testnets.json:23]: Suggestion: 
   "claim_stake": 0.00047,
    "claim_request_extension": 900,
    "claim_period": 900,
    "challenge_period_extension": 900

In order to better test challenges I suggest to lower the values to 900 (15 minutes)

**Review by fredo — CHANGES_REQUESTED** [review]

File: scripts/deployment/goerli-testnets.json (hunk 3 above)
Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json:35]: For Optimism on Goerli:
- L1 -> L2 = 5 minutes (https://community.optimism.io/docs/developers/bridge/messaging/#for-ethereum-l1-to-optimism-l2-transactions)
- L2 -> L1 = 5 minutes (https://community.optimism.io/docs/developers/bridge/messaging/#for-optimism-l2-to-ethereum-l1-transactions)

Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json:35]: Boba should be the same

File: scripts/deployment/goerli-testnets.json (hunk 2 above)
Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json:18]: By testing the L2 -> L1 messaging, i noticed waiting times between 30 - 60 minutes. 
Their ETH bridge L2 -> L1 takes ~ hours. 
L1 -> L2 is shorter, ~ 15-30 minutes.

So just to be safe i will use 5 hours here.

File: scripts/deployment/goerli-testnets.json (hunk 4 above)
Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json]: ✅

File: scripts/deployment/goerli-testnets.json (hunk 3 above)
Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json]: ✅

File: scripts/deployment/goerli-testnets.json (hunk 2 above)
Comment (GabrielBuragev) [reply scripts/deployment/goerli-testnets.json]: ✅

**Review by istankovic — APPROVED** [review]

**Pull request merged**